### PR TITLE
[otp] decouple CREATOR_SW_CFG_MANUF_STATE from SKU

### PR DESCRIPTION
**Pull request by anthonychen1251**
This commit decouples the `CREATOR_SW_CFG_MANUF_STATE` OTP field from the SKU.

Previously, the `CREATOR_SW_CFG_MANUF_STATE` field was used to identify the SKU. However, we now leverage Device ID for SKU identification, eliminating the need to use `CREATOR_SW_CFG_MANUF_STATE` for this purpose. We can now embed the first four bytes of the immutable ROM_EXT hash into the `CREATOR_SW_CFG_MANUF_STATE` field.

The following changes were made to implement this decoupling:

- The `CREATOR_SW_CFG_MANUF_STATE` field in the `otp_ctrl` data structure is now set to a fixed value, `CONST.MANUF_STATE.PERSONALIZED`.
- The `MANUF_STATE` enum in the `const.bzl` file has been updated to remove the `SIVAL` value and add a `PERSONALIZED` value.
- The `update_creator_manuf_state_data` function in the `ImmutableSectionProcessor.py` file has been updated to no longer require the `creator_manuf_state` argument.
- The `update_manifest_json.py` script has been updated to no longer require the `manuf_state_creator` argument.

Note: this depends on https://github.com/lowRISC/opentitan/pull/24948 and https://github.com/lowRISC/opentitan/pull/25087. Only review the last commit.
This PR addresses https://github.com/lowRISC/opentitan/issues/24610 partially.

### CODE REVIEW

File: util/design/update-manifest-json.py
Comment (timothytrippel) [inline util/design/update-manifest-json.py:42]: Shouldn't the manifest bazel rule already do this: https://cs.opensource.google/opentitan/opentitan/+/master:rules/manifest.bzl;l=81?q=manifest.bzl&ss=opentitan%2Fopentitan

**Review by timothytrippel — APPROVED** [review]
LGTM, thanks @anthonychen1251 ! Waiting for @cfrantz to review too before merging.

File: util/design/update-manifest-json.py
Comment (anthonychen1251) [reply util/design/update-manifest-json.py:42]: yes, the selector bit of `manuf_state_creator` field will be set by the manifest bazel rule if the `manuf_state_creator` attribute is provided.

The operation here is to handle the case that the `manuf_state_creator` attribute wasn't specified when generating the manifest json file. Alternatively,  we can assume that `manuf_state_creator` should be specified when generating the manifest json file and validate its presence by checking the selector bit here. Let me know which approach works best for you.

Comment (timothytrippel) [reply util/design/update-manifest-json.py:42]: Ah got it ok that makes sense to me. We can keep as is then.

**Comment by timothytrippel** [comment]
I rebased on #25087 to fix merge conflict. Will merge when this passes CI.

**Pull request merged**
